lib/puppet/util/tagging.rb

```diff
@@ -1,7 +1,7 @@
 require 'puppet/util/tag_set'
 
 module Puppet::Util::Tagging
-  ValidTagRegex = /\A[0-9A-Za-z_][0-9A-Za-z_:.-]*\Z/
+  ValidTagRegex = /\A[[:word:]][[[:word:]]:.-]*\Z/
```

Comment:
I'm not overly fond of [:word:] since it's a non-standard construct. Documentation states that it includes everything in the general Unicode category 'Mark' and that doesn't seems right. Also, I think there are one pair of brackets too many in the expression. How about this?
```
/\A[[:alnum:]_][[:alnum:]_:.-]*\z/
```